netket/graph/lattice.py

```diff
@@ -104,323 +111,405 @@ def get_edges(atoms_positions, cutoff, distance_atol=cutoff_tol):
+            or not all(isinstance(b, bool) for b in pbc)
+        ):
+            raise ValueError(
+                "pbc must be either a boolean or a sequence of booleans with length equal to "
+                "the lattice dimenion"
+            )
+        else:
+            return _np.asarray(pbc, dtype=bool)
 
-    def planar_rotations(self, period, axes=[0, 1]) -> PermutationGroup:
+    # Properties
+    # ------------------------------------------------------------------------
+    @property
+    def basis_vectors(self):
         """
-        Returns PermutationGroup corresponding to rotations about specfied axes with specified period
+        Basis vectors of the lattice
+        """
+        return self._basis_vectors
 
-        Arguments:
-            period: Period of the rotations
-            axes: Axes that define the plane of the rotation
+    @property
+    def site_offsets(self):
+        """
+        Position offsets of sites in the unit cell
         """
+        return self._site_offsets
 
-        perm = self.rotation_perm(period, axes)
-        rotations = [PlanarRotation(perm, n) for n in range(1, period)]
+    @property
+    def ndim(self):
+        """Dimension of the lattice"""
+        return self._ndim
 
-        return PermutationGroup([Identity()] + rotations, degree=self.n_nodes)
+    @property
+    def sites(self) -> Sequence[LatticeSite]:
+        return self._sites
 
-    def basis_translations(self) -> PermutationGroup:
+    @property
+    def positions(self) -> PositionT:
         """
-        Returns PermutationGroup corresponding to translations by basis vectors
+        Real-space positions of all lattice sites
         """
+        return self._positions
 
-        translations = product(*[range(i) for i in self.extent])
-        next(translations)
+    @property
+    def basis_coords(self) -> CoordT:
+        """
+        basis coordinates of all lattice sites
+        """
+        return self._basis_coords
 
-        perms = self.translation_perm()
-        translations = [Translation(perms, i) for i in translations]
+    # Site lookup
+    # ------------------------------------------------------------------------
+    def _to_integer_position(self, positions: PositionT) -> int:
+        frac_positions = _np.matmul(positions, self._inv_dims) % 1
+        return _np.around(frac_positions * 10 ** tol_digits).astype(int) % (
+            10 ** tol_digits
+        )
 
-        return PermutationGroup([Identity()] + translations, degree=self.n_nodes)
+    def id_from_position(self, position: PositionT) -> int:
+        """
+        Return the id for a site with given position.
+        Throws a KeyError if no corresponding site is found.
+        """
+        int_pos = HashableArray(self._to_integer_position(position))
+        try:
+            return self._int_position_to_site[int_pos]
+        except KeyError as e:
+            raise KeyError(f"No site found for position={position}") from e
```

Comment:
Could we offer a vectorised lookup for 2D arrays? I know it would only be a loop, but at least one would spare the function calls when matching the entire lattice...

Reply:
Yeah, why not, I can do that. Maybe this should return `None` instead of throwing an exception if there is not site at the position (especially when you can pass multiple inputs).

Reply:
Hmm, I'm open for either. My use case would be matching sites when acting on them with some symmetry, so in that case it's good if it dies from a single unmatched entry. Of course checking for `None` works too